perftests/standalone-enc/freebsd_flamegraph.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+
+# To see only profile data from pipe_enc_thread:
+#       ./freebsd_flamegraph.sh
+#
+# To see all profile data:
+#       ./freebsd_flamegraph.sh --all
+
+set -e -o noclobber -o nounset
+
+outfilename="bench"
+
+# Should we keep all the profile data?
+KEEP_ALL=0
+if [ "$#" -eq "1" ]; then
+	# The POSIX sh does not do lazy evaluation, so this needs to be a
+	# separate `if` statement.
+	if [ "$1" = "--all" ]; then
+		KEEP_ALL=1
+	fi
+fi
+
+# Set dtrace command; if we're not root, we must have doas(1) configured.
+if [ "$(id -u)" -eq "0" ]; then
+	DTRACE_CMD="dtrace"
+else
+	DTRACE_CMD="doas /usr/sbin/dtrace"
+fi
+
+# Clear previous data
+rm -f "${outfilename}.stacks" "${outfilename}.folded" "${outfilename}.svg"
+
+# Get profile data and consolidate it.
+${DTRACE_CMD}						\
+	-x ustackframes=100				\
+	-n 'profile-97 { @[ustack()] = count(); }'	\
```

Comment:
I would use `profile-997` rather than `profile-97` in order to get more data.  Also perhaps add an option to `test_standalone_enc` to increase the amount of data it handles; the current 100 MB doesn't allow for many data points to be gathered.

You should also ask dtrace to filter the profiling hits: We only want to record when execname is "test_standalone_enc" and arg1 is nonzero.  Otherwise we'll gather data about other processes on the system and time spent in the kernel.  Add `/execname == "test_standalone_enc" && arg1/` after `profile-N`.

Reply:
Doesn't the `-c` take care of this?  I'm not seeing info about other processes?  Or maybe that's just because I'm only running vim at the same time as profiling, and that has minimal resource usage?

Since it's only counting `ustack()` I thought it wouldn't be picking up kernel stuff.

But ok, I'll do the `/execname` thing.

Reply:
Granted, the man-page makes no claims about `-c` only gathering info about the specified command, so ok.

Reply:
The long "tail" on the left of your SVG is dtrace reporting all the times the probe fired and the kernel was running the idle thread.